service/src/main/scala/com/yahoo/maha/service/MahaService.scala

```diff
@@ -657,4 +662,139 @@ object MahaServiceConfig {
     result
   }
 
-}
\ No newline at end of file
+}
+
+
+case class DynamicMahaServiceConfig(dynamicProperties: Map[String, DynamicPropertyInfo], context: MahaServiceConfigContext, registry: Map[String, RegistryConfig], mahaRequestLogWriter: MahaRequestLogWriter, curatorMap: Map[String, Curator])
```

Comment:
can you more test to test that how dynamic configs are applied in the case of existing pending execution ?

Reply:
and also if dynamic config is failed to apply, what happens, test should include all cases

Reply:
This is just the creation of dynamic config, not its use. Will add more logic and tests incrementally.